parser_library/src/processing/op_code.cpp

```diff
@@ -14,43 +14,48 @@
 
 #include "op_code.h"
 
+#include <assert.h>
+
 #include "context/instruction.h"
 
 namespace hlasm_plugin::parser_library::processing {
 
-inline unsigned char get_reladdr_bitmask(context::id_index id)
+inline unsigned char get_reladdr_bitmask(
+    context::id_index id, const hlasm_plugin::parser_library::context::instruction_sets& instructions)
 {
     if (!id || id->empty())
         return 0;
 
-    if (auto p_instr = context::instruction::find_machine_instructions(*id))
+    if (auto p_instr = instructions.find_machine_instructions(*id))
         return p_instr->reladdr_mask().mask();
 
-    if (auto p_mnemo = context::instruction::find_mnemonic_codes(*id))
+    if (auto p_mnemo = instructions.find_mnemonic_codes(*id))
         return p_mnemo->reladdr_mask().mask();
 
     return 0;
 }
 
 // Generates value of L'* expression
-unsigned char processing_status_cache_key::generate_loctr_len(context::id_index id)
+unsigned char processing_status_cache_key::generate_loctr_len(
+    context::id_index id, const hlasm_plugin::parser_library::context::instruction_sets& instructions)
 {
     if (id && !id->empty())
     {
-        if (auto p_instr = context::instruction::find_machine_instructions(*id))
+        if (auto p_instr = instructions.find_machine_instructions(*id))
             return static_cast<unsigned char>(p_instr->size_in_bits() / 8);
 
-        if (auto p_mnemo = context::instruction::find_mnemonic_codes(*id))
+        if (auto p_mnemo = instructions.find_mnemonic_codes(*id))
             return static_cast<unsigned char>(p_mnemo->instruction()->size_in_bits() / 8);
     }
     return 1;
 }
 
-processing_status_cache_key::processing_status_cache_key(const processing_status& s)
+processing_status_cache_key::processing_status_cache_key(
+    const processing_status& s, const hlasm_plugin::parser_library::context::instruction_sets& instructions)
     : form(s.first.form)
     , occurence(s.first.occurence)
     , is_alias(s.second.type == context::instruction_type::ASM && s.second.value && *s.second.value == "ALIAS")
-    , loctr_len(generate_loctr_len(s.second.value))
-    , rel_addr(get_reladdr_bitmask(s.second.value))
+    , loctr_len(generate_loctr_len(s.second.value, instructions))
```

Comment:
```suggestion
    , loctr_len(s.second.type == context::instruction_type::MACH?0:generate_loctr_len(s.second.value, instructions))
```
or pass `s.second.type` as arg?